org-ql-view.el

```diff
@@ -1093,30 +1108,37 @@ representation `org-ql-view-buffers-files' is returned."
   "Return BUFFERS-FILES expanded to a list of files or buffers.
 The counterpart to `org-ql-view--contract-buffers-files'.
 This always returns a list of string values."
-  (-->
-      (-map (lambda (buffer-file)
-                (pcase-exhaustive buffer-file
-                  ("all" (--select (equal (buffer-local-value 'major-mode it) 'org-mode)
-                                   (buffer-list)))
-                  ("org-agenda-files" (org-agenda-files))
-                  ("org-directory" (org-ql-search-directories-files))
-                  ((or "" "buffer")
-                   (current-buffer))
-                  ((pred bufferp) (list buffer-file))
-                  ;; A single filename.
-                  ((pred stringp) (list buffer-file))
-                  (_ (error (format "Value %s is not a valid buffer/file" buffer-file)))))
-            (-list buffers-files))
-    -flatten -non-nil
-    ;; expanding all file-buffers to file names to avoid duplicate entries being formed
-    (-map (lambda (buffer-file)
-            (if (bufferp buffer-file)
-                (--if-let (buffer-file-name buffer-file)
-                    it
-                  (buffer-name buffer-file))
-              buffer-file))
-          it)
-    -uniq))
+  (let ((expanded-buffers-files
+         (--> buffers-files
+           -list -non-nil
+           (-map (lambda (buffer-file)
+                   (pcase-exhaustive buffer-file
+                     ("all" (--select (equal (buffer-local-value 'major-mode it) 'org-mode)
+                                      (buffer-list)))
+                     ("org-agenda-files" (org-agenda-files))
+                     ("org-directory" (org-ql-search-directories-files))
+                     ((or "" "buffer")
+                      (current-buffer))
+                     ((or (pred bufferp)
+                          ;; A single filename.
+                          (pred stringp))
+                      buffer-file)
+                     (_ (error (format "Value %s is not a valid buffer/file" buffer-file)))))
+                 it))))
+    (--> expanded-buffers-files
+      -flatten
+      ;; removing `nil' again as some values have been expanded.
+      -non-nil
+      ;; Expanding all buffers to file names or buffer names to remove duplicate entries.
+      (--map
+       (pcase-exhaustive it
+         ((pred bufferp) (or (buffer-file-name it)
+                             (buffer-name it)))
+         ;; Any values at this point should be a buffer or string.
+         ;; Testing for string anyways.
+         ((pred stringp) it))
+       it)
+      -uniq)))
```

Comment:
This is still a very long, narrow form that is very hard to follow.  Maybe you could rewrite it in a functional way using `cl-labels`.